core/src/main/scala/pickling/PBuilderReader.scala

```diff
@@ -19,6 +19,10 @@ trait Hintable {
 trait PBuilder extends Hintable {
   def beginEntry(picklee: Any): PBuilder
   def putField(name: String, pickler: PBuilder => Unit): PBuilder
+  def putDynamicFieldNames(names: List[String])(implicit namesPickler: SPickler[List[String]]): PBuilder =
```

Comment:
is there benefit to making this an iterable or something more abstract, to try to avoid copying if someone's passing in `myMap.keys()` here? in theory you could iterate the keys with some kind of view instead of a new data structure.